[#956] Convert documentation to MarkBind and deploy using Surge

jamessspanggg opened the pull request:
Fixes #956

```
Our User Guide and Development Guide is using markdown.

Let's include the deployment of markbind docs to surge, while adding
the github status that redirects to the link of the deployed site.
```

- [x] Fixed broken links
- [x] Created side navigation
- [x] Standardised naming of deployed sites.
- [x] Use existing README as home page. (can be changed to a specific home page in the future)

Tejas2805 commented:
Did you build a new markbind site ?

You can convert our UG and DG into a site directly by running - `markbind init --convert`

jamessspanggg commented:
@damithc managed to deploy markbind in addition to our dashboard with some help from @alyip98. Can view the github status `docs/surge/deploy/1197-pr` to view the generated markbind site.

> You can convert our UG and DG into a site directly by running - markbind init --convert

Should we convert our UG and DG straight away using `markbind init --convert`?

Tejas2805 commented:
> Should we convert our UG and DG straight away using `markbind init --convert`?

I think it will be better. Cause once this is done. We just need to re-structure and format the site.

![image](https://user-images.githubusercontent.com/35946746/77894797-62730a00-72a8-11ea-9a73-2e356c89d334.png)

damithc commented:
> Yes I used the default generated site for testing purposes. Should we convert our UG and DG straight away using `markbind init --convert`?

You can give it a try. I don't think anybody actually used that feature ever since it was released :-p

Tejas2805 commented:
> You can give it a try. I don't think anybody actually used that feature ever since it was released :-p

Haha. PR #1113 uses it. Works well. :)

jamessspanggg commented:
> You can give it a try. I don't think anybody actually used that feature ever since it was released :-p

Ok will do. But is this workflow fine? i.e. deploying to separate surge sites, and using separate github status to redirect to corresponding link.

damithc commented:
> Ok will do. But is this workflow fine? i.e. deploying to separate surge sites, and using separate github status to redirect to corresponding link.

Yes, it's fine.

jamessspanggg commented:
@damithc I have converted our existing docs to markbind. Can I get some feedback from you?

Note that I have done the following:
- Fixed existing broken links in documentation
- Moved initial navigation within the page to the side nav
- Add side nav for pages without them
- Used the existing README as home page. But I think we need to change this to a custom home page instead of using the same as README. But perhaps this can be done in another PR?

damithc commented:
> @damithc I have converted our existing docs to markbind. Can I get some feedback from you?

Looks good 👍  At this stage, it's good enough to just convert to MarkBind; no need for any drastic changes to the content. We can improve the content later.

jamessspanggg commented:
> Looks good 👍 At this stage, it's good enough to just convert to MarkBind; no need for any drastic changes to the content. We can improve the content later.

Sure. Will proceed with the reviews.

jamessspanggg commented:
@reposense/devs need your reviews here to ensure if there's any mistakes such as broken links/missing content etc.

Tejas2805 commented:
> Looks good 👍 At this stage, it's good enough to just convert to MarkBind; no need for any drastic changes to the content. We can improve the content later.

I have a few suggestions and stuff on how to improve the content and organization for the purpose of the website, since I have already did it on my side. But will just keep them for now, and create issues and work on them later as mentioned by Prof. 

For now,  have reviewed based on current stage.

Tejas2805 commented on docs/_markbind/navigation/ugNavigation.md:
```diff
@@ -0,0 +1,24 @@
+    * [`group-config.csv`]({{baseUrl}}/UserGuide.html#group-config-csv)
+* [Analyzing Multiple Repos]({{baseUrl}}/UserGuide.html#analyzing-multiple-repos)
+* [Quickstart RepoSense with Netlify]({{baseUrl}}/UserGuide.html#quickstart-reposense-with-netlify)
+* [Using Travis-CI to automate publishing of the report to GitHub Pages]({{baseUrl}}/UserGuide.html#automating-publishing-of-the-report-to-github-pages)
```
Should be `Automate publishing of report to Github Pages`

jamessspanggg commented:
> I have a few suggestions and stuff on how to improve the content and organization for the purpose of the website, since I have already did it on my side. But will just keep them for now, and create issues and work on them later as mentioned by Prof.

Hmm do u mean the organization of the UI? or the organization of the markbind components (i.e. how to place the folders, which components should be in which folders etc) ? If it is the latter, perhaps u can suggest here?

jamessspanggg commented:
@Tejas2805 some questions while looking through your pr. Asking because I'm not that familiar with MB.

I noted that you committed the `_site` folder in ur PR, isn't the `_site` folder dynamically generated when u run `markbind build`? is there a specific reason that you are committing it?

I also noted that the markbind project is built in the root directory instead of in `docs`. Is there a reason behind that?

Tejas2805 commented:
> I noted that you committed the _site folder in ur PR, isn't the _site folder dynamically generated when u run markbind build? is there a specific reason that you are committing it?

I was trying it out for the first time too. So just committed it for once. No need to commit it.

> I also noted that the markbind project is built in the root directory instead of in docs. Is there a reason behind that?

MarkBind recommends running the `convert` command in the root directory so I went ahead with it. But since, we don't need it for the whole directory but just docs, we should build it in docs, like done in this PR.

Tejas2805 commented on docs/DeveloperGuide.md:
```diff
@@ -280,20 +257,20 @@ The summary view acts as the main report which shows the various calculations. <
+- [**main.js**](#main-main-js) - main controller that pushes content into different modules
+- [**api.js**](#data-loader-api-js) - loading and parsing of the report content
+- [**v_summary.js**](#summary-view-v-summary-js) - module that supports the summary view
+- [**v_authorship.js**](#authorship-view-v-authorshipjs) - module that supports the authorship tab view
```
`authorship-view-v-authorship-js` is better?

jamessspanggg replied:
yep, thanks. Missed that

jamessspanggg commented:
> Maybe you can add a title for the other pages too? Not something important for this PR though. So, upto you to include it or not.

Have added the titles and fixed the broken link.

yong24s commented on .travis.yml:
```diff
@@ -62,5 +62,9 @@ after_success:
+      npm install -g markbind-cli
+      cd docs
+      markbind build
+      cd ..
```
Optional suggestion, not sure but should work on Travis-CI :thinking::

```diff
- cd docs
- markbind build
- cd ..
+ (cd docs && markbind build)
```
This way you don't need to `cd` back as only the subshell's `pwd` is modified.

jamessspanggg replied:
seems to work fine!

jamessspanggg commented:
@ang-zeyu can i get ur review on this? since you are working on markbind, you might be clearer on how things work.

Tejas2805 commented:
Ya, had the same comment too. But since the Prof said looks good for now, I let it be for the time being. 

Let’s re-organise stuff in another PR. And just deploy for once, as recommended by Prof

jamessspanggg commented:
> You could use the auto generated [page nav](https://markbind.org/userGuide/tweakingThePageStructure.html#page-navigation-menus) for intra page navigation, and two site navs for userGuide and devGuide inter page navigation perhaps.

I think i can do it in this PR as currently I have a specific siteNav for each page. Doesn't make sense to add and delete all of those again in the future.

ang-zeyu commented on docs/_markbind/navigation/mainNav.md:
```diff
@@ -0,0 +1,52 @@
+    * [Automate publishing of the report to GitHub Pages]({{baseUrl}}/UserGuide.html#automating-publishing-of-the-report-to-github-pages)
+    * [FAQ]({{baseUrl}}/UserGuide.html#faq)
+
+* <span class="lead">**Developer Guide**</span>
```
Hmm I meant two completely separate site navs.
E.g. if you're on a user guide page, only the expanded user guide sitenav would show

Example of the userGuide site nav in markbind:
![Untitled](https://user-images.githubusercontent.com/3306138/78424561-0ee62f00-76a1-11ea-8134-1571e79b95dc.png)


The links in the sitenav seem to reflect the pagenav's links as well, you could instead include all the **pages** of the userGuide under the userGuide sitenav for inter page navigation, which is its intended use case. ( there aren't many pages for now as well but would be nice to have the structure layed out so we could always split it up later easily )

jamessspanggg replied:
Done. There were only two inter pages each for ug and dg.

jamessspanggg replied:
Is there a way such that if i only add a pageNav (without the siteNav), the content in the center won't be pushed towards the left of the entire page?

ang-zeyu replied:
not sure if it works, but you could try an empty an empty sitenav with just the `<navigation>` tag

jamessspanggg replied:
yep this works fine. Thanks so much for your help!

ccyccyccy commented:
Seems good. However, the left nav bar disappears when I click on any of the links there. Is that intended?

jamessspanggg commented:
> Seems good. However, the left nav bar disappears when I click on any of the links there. Is that intended?

Yes the left nav currently is only available for UG and DG. The left nav represents inter page links of those two pages.

ccyccyccy commented:
> > Seems good. However, the left nav bar disappears when I click on any of the links there. Is that intended?
> 
> Yes the left nav currently is only available for UG and DG. The left nav represents inter page links of those two pages.

Ok got it.

damithc commented:
> Yes the left nav currently is only available for UG and DG. The left nav represents inter page links of those two pages.

I think the siteNave (i.e., left side) should be there in all pages.
* TopNav: provides a site map, but only top-level important nodes
* SiteNav (left side): provides a site map, including some mid-level nodes. Should be same for all pages.
* PageNav(right side): provides a map to the current page only

You can follow the example of se-edu/se-book

jamessspanggg commented:
> > Yes the left nav currently is only available for UG and DG. The left nav represents inter page links of those two pages.
> 
> I think the siteNave (i.e., left side) should be there in all pages.
> 
> * TopNav: provides a site map, but only top-level important nodes
> * SiteNav (left side): provides a site map, including some mid-level nodes. Should be same for all pages.
> * PageNav(right side): provides a map to the current page only
> 
> You can follow the example of se-edu/se-book

Ok noted.

anubh-v commented on docs/UserGuide.md:
```diff
@@ -163,7 +149,7 @@ Repo owners can provide the following additional information to RepoSense using
 * the display of an author
 
-To use this feature, add a `_reposense/config.json` to the root of your repo using the format in the example below ([another example](../_reposense/config.json)) and **commit it** (reason: RepoSense can see committed code only):
+To use this feature, add a `_reposense/config.json` to the root of your repo using the format in the example below ([another example](https://github.com/reposense/RepoSense/blob/master/_reposense/config.json)) and **commit it** (reason: RepoSense can see committed code only):
```
Is there a reason we are linking to the `config.json` on master branch of the repo, rather than using a relative link to `config.json` ? (so that we can see the `config.json` in the same branch as this document)

jamessspanggg replied:
The reason why I changed it to master branch is because the markbind project is built within the `docs` folder. I couldn't find a way to access resources that are out of the `docs` folder.

@ang-zeyu is there a workaround for this?

damithc replied:
I think absolute path is the only way as this is a link outside of the site as far as the documentation site is concerned. The html pages are deployed onto a different branch (i.e., gh-pages) anyway, irrespective of the source branch.

anubh-v commented:
Currently, our RepoSense dashboard has a link (at the bottom of the page) to the User Guide associated with that version of RepoSense. 
Would we now be directing this link to the MarkBind docs?

jamessspanggg commented:
> Would we now be directing this link to the MarkBind docs?

No, that would only direct to the corresponding `UserGuide.md` in GitHub.

jamessspanggg commented:
@damithc require your final review on this.

jamessspanggg commented:
@eugenepeh @fzdy1914 ready to merge

fzdy1914 commented:
Should we include publish guide in the main nav bar?

jamessspanggg commented:
> Should we include publish guide in the main nav bar?

I did not place it as in the main nav bar as I think the publish guide is not considered as one of the main sections. But it is available to access within the user guide.

jamessspanggg commented:
@damithc should we include the [publishing guide](https://github.com/reposense/RepoSense/blob/master/docs/PublishingGuide.md) to the siteNav? Currently it is accessible within the user guide only.

damithc commented:
> @damithc should we include the [publishing guide](https://github.com/reposense/RepoSense/blob/master/docs/PublishingGuide.md) to the siteNav? Currently it is accessible within the user guide only.

Should be fine for now.

fzdy1914 commented:
Will merge this, however, should we also include an instruction for how to run markbind locally in DG?

jamessspanggg commented:
> Will merge this, however, should we also include an instruction for how to run markbind locally in DG?

Added instructions to install and run markbind in DG.
